flower/utils/results/stores/redis.py

```diff
@@ -23,7 +22,7 @@ def results_by_timestamp(self, limit: int | None = None, reverse: bool = True) -
         for key in self.backend.client.scan_iter(
             match=task_key_prefix + ("*" if isinstance(task_key_prefix, str) else b"*")
         ):
-            result_data: dict[str, Any] = json.loads(self.backend.client.get(key))
+            result_data: dict[str, Any] = self.backend.decode_result(self.backend.client.get(key))
```

Comment:
This fixes an earlier issue with our Redis result store. Simply invoking the `json.loads()` method is incomplete, because it will not use the custom decoding logic known to Celery's deserialization methods. So we should instead use something within Celery-land to perform decoding.

Reply:
Cool, makes sense!